spec/controllers/admin/channels_controller_spec.rb

```diff
@@ -15,22 +15,57 @@
     let(:channel)         { create(:channel, recurring: true, custom_submit_text: 'custom_text') }
     let(:updated_channel) { Channel.find(channel.id) }
     let(:category)        { create(:category) }
+    let(:updated_channel) { Channel.find channel.id }
     let(:channel_attributes) do
       {
-        category_id: category.id,
-        name: 'new_name',
-        email: 'email_channel_123@foo.bar',
-        description: 'Foo description bar',
-        permalink: 'foo_permalink_test_bar',
-        recurring: false,
-        custom_submit_text: 'update_custom_text'
+        category_id:        category.id,
+        custom_submit_text: 'update_custom_text',
+        description:        'Foo description bar',
+        email:              'email_channel_123@foo.bar',
+        name:               'new_name',
+        permalink:          'foo_permalink_test_bar',
+        recurring:          false,
+        visible:            false
       }
     end
 
-    before do
-      put(:update, id: channel.id, channel: channel_attributes, locale: :pt)
+    context "when the channel is recurring" do
+      let(:channel) { create(:channel, :recurring) }
+      let(:error_message) { I18n.t('admin.channels.messages.recurring.error.update') }
+
+      context "and it has projects associated" do
+        before do
+          create(:project, channels: [channel])
+
+          put(:update, id: channel.id, channel: channel_attributes, locale: :pt)
+        end
+
+        it "should not have updated the channel" do
```

Comment:
tenta não usar should nos testes. escreve algo como it "updates the channel"
na terceira pessoa do presente